### ISSUE TITLE
could not compile `wasmedge-sys` when trying to run `hello world` example

### ISSUE THREAD

**Issue by omgirok** [WasmEdge/WasmEdge#2147]
Running into a compile error when following the documentation here:
https://wasmedge.org/book/en/write_wasm/rust.html#hello-world

Steps taken:
1) copy-paste hello world rust code into `main.rs`
2) `rustup target add wasm32-wasi`
3) `cargo build --target wasm32-wasi`

3rd step is when i started running into problems:
**1st issue:**
```
error: failed to run custom build command for `wasmedge-sys v0.12.2`

Caused by:
  process didn't exit successfully: `C:\Users\orion.ou\rust_project\wasm-test\target\debug\build\wasmedge-sys-cdb407e04287e619\build-script-build` (exit code: 101)
  --- stdout
  cargo:warning=[wasmedge-sys] Failed to locate lib_dir, include_dir, or header.

  --- stderr
  thread 'main' panicked at '[wasmedge-sys] Failed to locate the required header and/or library file.
```
fixed by:
```
$env:WASMEDGE_INCLUDE_DIR="C:\Users\orion.ou\.wasmedge\include\";
$env:WASMEDGE_LIB_DIR="C:\Users\orion.ou\.wasmedge\lib\"
```
then try to run `cargo build --target wasm32-wasi` again

**2nd issue:**
```
thread 'main' panicked at 'Unable to find libclang: "couldn't find any valid shared libraries matching: ['clang.dll', 'libclang.dll'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])
```
fixed by:
installing LLVM 

then try to run `cargo build --target wasm32-wasi` again

**3rd issue (stuck here):**
 ```
(dev) orion.ou ~\wasm-test [01:31PM]> cargo build --target wasm32-wasi
   Compiling wasmedge-sys v0.12.2
error[E0425]: cannot find function `path_to_cstring` in module `utils`
  --> C:\Users\orion.ou\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.12.2\src\loader.rs:85:29
   |
85 |         let c_path = utils::path_to_cstring(file.as_ref())?;
   |                             ^^^^^^^^^^^^^^^ not found in `utils`

error[E0425]: cannot find function `path_to_cstring` in module `utils`
   --> C:\Users\orion.ou\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.12.2\src\vm.rs:136:27
    |
136 |         let path = utils::path_to_cstring(file.as_ref())?;
    |                           ^^^^^^^^^^^^^^^ not found in `utils`

error[E0425]: cannot find function `path_to_cstring` in module `utils`
   --> C:\Users\orion.ou\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.12.2\src\vm.rs:616:27
    |
616 |         let path = utils::path_to_cstring(path.as_ref())?;
    |                           ^^^^^^^^^^^^^^^ not found in `utils`

error[E0308]: mismatched types
   --> C:\Users\orion.ou\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.12.2\src\types.rs:210:54
    |
210 |             ctx: unsafe { ffi::WasmEdge_ValueGenV128(val) },
    |                           -------------------------- ^^^ expected struct `int128_t`, found `i128`
    |                           |
    |                           arguments to this function are incorrect
    |
note: function defined here
   --> C:\Users\orion.ou\rust_project\wasm-test\target\wasm32-wasi\debug\build\wasmedge-sys-69dfc6e19371869f\out/wasmedge.rs:973:12
    |
973 |     pub fn WasmEdge_ValueGenV128(Val: int128_t) -> WasmEdge_Value;
    |            ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> C:\Users\orion.ou\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.12.2\src\types.rs:217:18
    |
216 |     pub fn to_v128(&self) -> i128 {
    |                              ---- expected `i128` because of return type
217 |         unsafe { ffi::WasmEdge_ValueGetV128(self.ctx) }
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i128`, found struct `int128_t`

Some errors have detailed explanations: E0308, E0425.
For more information about an error, try `rustc --explain E0308`.
The following warnings were emitted during compilation:

warning: [wasmedge-sys] libwasmedge found in C:\Users\orion.ou\.wasmedge\lib\

error: could not compile `wasmedge-sys` due to 5 previous errors
```

I have WasmEdge 0.11.2
LLVM 12.0.0
Cargo.toml has wasmedge-sdk = "0.7.1"

**Comment by apepkuss** [comment]
Hi @omgirok,

Thanks for your report!

The example code in https://wasmedge.org/book/en/write_wasm/rust.html#hello-world does not have dependencies on `wasmedge-sdk` and `wasmedge-sys` crate, so the issue can be solved by removing `wasmedge-sdk = "0.7.1"` from Cargo.toml. After that, you can try compiling it again.

If you need to use `wasmedge-sdk` in your project, please refer to [WasmEdge Installation And Uninstallation](https://wasmedge.org/book/en/quick_start/install.html#wasmedge-installation-and-uninstallation) to deploy `WasmEdge Runtime library` first. If the library is deployed in the default directory, `wasmedge-sdk` will find it automatically; otherwise, you can set `WASMEDGE_INCLUDE_DIR` and `WASMEDGE_LIB_DIR` environment variables to specify the locations.

For the third issue, we checked the relevant code. According to the result of our CI workflow for `Windows` platform, the `utils::path_to_cstring` method works correctly. Could you please share with us the entire example code you're using? so that we can try reproducing the issue with your code. Thanks a lot!

**Comment by hydai** [comment]
This issue was closed because it has been inactive. Please feel free to re-open it if you still have the same issue. Thanks.
